server/views/cms/cms.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from flask import jsonify, request
+import requests
+import html
+
+from server import app
+from server.util.request import api_error_handler
+
+logger = logging.getLogger(__name__)
+BASE_URL = app.config['PAYLOAD_API_URL']
+API_KEY = app.config['PAYLOAD_API_KEY']
+
+if not BASE_URL or not API_KEY:
+    error_message = 'Configuration error: PAYLOAD_API_URL and/or PAYLOAD_API_KEY are missing.'
+    logger.error(error_message)
+    raise RuntimeError(error_message)
+
+BASE_URL = BASE_URL.rstrip('/')
+
+@app.route('/api/cms/pages', methods=['GET'])
+@api_error_handler
+def api_fetch_page_content():
+    application_name = html.escape(request.headers.get('cs-app') or '')
+
+    if not application_name:
```

Comment:
```suggestion
    application_name = html.escape(request.headers.get('cs-app') or '')
    if not application_name:
```

`'cs-app'` should at least be a constant if not a config/env.

Reply:
@kilemensi  It might not be a good idea to have it as a `config` since this file is only read by Flask, and I believe we have one backend that is accessed by multiple font-end apps so we don't really need to tie the backend to a specific app.

If we set it as an `.env `variable,  we would will still need to pass the value to the server side either as a `header` or `url parameter`